feat: old content warning

ericswpark opened the pull request:
WIP, needs styling on the alert div. Basic functionality works

Closes #250

ericswpark commented:
Forgot to mention but thanks to https://brunty.me/post/display-message-on-old-hugo-entries/ for the original code.

@lxndrblz @SchabrechtsK please let me know if this implementation is okay and please help me with styling. I'm not sure what will look good. Right now the text is inserted right above the content, without any visual distinction which might look odd.

lxndrblz commented:
@ericswpark Thanks! I'll take care of the CSS styling.

ericswpark commented:
I changed the implementation so it now displays the age of the post rather than the duration cutoff, which looks better IMHO at conveying the age of the post. Also moved the string to i18n.

Should we consider allowing users to customize the outdated content message shown? Or should we just display the generalized outdated statement?

lxndrblz commented:
@ericswpark I would leave it a default message and not customizable. 

Personally, I would change the wording to say something like "over xxx days" old, because the problem of this static approach is that if a user does not rebuild their sites daily, the information would be incorrect.

lxndrblz commented:
@ericswpark Just pushed some basic CSS styling:

![image](https://user-images.githubusercontent.com/12200062/141161862-a5537385-fb14-494b-a429-afe8b32872b1.png)

What do you think?

ericswpark commented:
@lxndrblz 

> Personally, I would change the wording to include the value set in the old content duration and emphasis the "over", because the problem of this static approach is that if a user does not rebuild their sites daily, the information would be incorrect.

I haven't even considered that! I'll change it back to the original wording.

> ...
> ![image](https://user-images.githubusercontent.com/12200062/141161862-a5537385-fb14-494b-a429-afe8b32872b1.png)
> 
> What do you think?

Looks good. Maybe we can make it yellow-colored to better distinguish it as a warning on the page.

ericswpark commented:
@lxndrblz here's the alert with the new changes:

![image](https://user-images.githubusercontent.com/13326074/141183575-ab53fa22-5aec-4cb7-a9c2-4385f0632172.png)

~~The "!" indicator wasn't centered vertically in the circle and I couldn't find a way to center it with padding so I just changed the height to center the indicator~~ I totally meant to make it sideways-oval shaped; it looks better right?

Also added fix for redefined post section name.

ericswpark commented on assets/css/style.css:
```diff
@@ -1041,6 +1042,7 @@ a.btn {
   padding: 0.5rem;
   height: 1.5rem;
   width: 2.5rem;
+  height: 2.5rem;
```
@lxndrblz was this added by mistake? Now there are two height attributes

lxndrblz replied:
No, that was by mistake. I forgot to remove the lower value.

ericswpark replied:
Hmm, then should we find a way to vertically center the "!" mark in the circle?

lxndrblz replied:
It should be resolved by setting the height identical to the width. Or am I am missing something?

ericswpark replied:
Actually never mind, I thought it wasn't centered vertically in the screenshot but it is. I wonder if I only saw it on my machine but I remember the "!" symbol was not properly vertically centered in the circle.

lxndrblz commented:
We should also add the remaining translations before this goes live.

ericswpark commented:
@lxndrblz wouldn't it be better to make it a pinned issue? Translators can't tack on translation commits to this PR unless they are a maintainer of the repository and getting all the translations in will take a lot of time unless we can contact all the previous translators. Since the feature is disabled by default I think translations will trickle in as people enable it and see that there is no translated string.

lxndrblz reviewed (approved):
@ericswpark LGTM! Thanks for all your help.

SchabrechtsK commented:
Very nice! Thanks for the hard work @ericswpark !   
Personal circumstances prevented me from working at all. Great to see it in here 😄
